fixit/rules/no_namedtuple.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Sequence, Tuple
+
+import libcst as cst
+from libcst import MaybeSentinel, ensure_type, parse_expression
+from libcst.metadata import QualifiedName, QualifiedNameProvider, QualifiedNameSource
+
+from fixit import CstLintRule, InvalidTestCase as Invalid, ValidTestCase as Valid
+
+
+class NoNamedTupleRule(CstLintRule):
+    """
+    Enforce the use of `dataclasses.dataclass` decorator instead of NamedTuple for cleaner customization and
+    inheritance. It supports default value, combining fields for inheritance, and omitting optional fields at
+    instantiation. See PEP 557 (https://www.python.org/dev/peps/pep-0557).
+    """
+
+    MESSAGE: str = "Instead of NamedTuple, consider using the @dataclass decorator from dataclasses instead."
```

Comment:
```suggestion
    MESSAGE: str = "Instead of NamedTuple, consider using the @dataclass decorator from dataclasses instead for simplicity, efficiency and consistency."
```